pkg/tsdb/sqleng/sql_engine.go

```diff
@@ -288,6 +288,10 @@ func (e *dataPlugin) executeQuery(query plugins.DataSubQuery, wg *sync.WaitGroup
 				continue
 			}
 
+			if t := frame.Fields[i].Type(); t == data.FieldTypeString || t == data.FieldTypeNullableString {
```

Comment:
Maybe this should be handled in `newProcessCfg`? Change `metricIndex` to `metricIndices` perhaps? On the other hand the logic for finding one metricIndex might no longer be applicable with your suggested changes?